### ISSUE TITLE
Ubuntu does not provide Modules/_hacl/libHacl_Hash_SHA2.a

### ISSUE THREAD

**Issue by nelgin** [InterNetNews/inn#302]
./configure fails on Ubuntu 24.04

```
configure:15775: checking for Py_Initialize
configure:15775: gcc -o conftest -g -O2 -I/usr/include/python3.12   conftest.c -L/usr/lib/x86_64-linux-gnu -lpython3.12 -ldl -lm -lm -lm -lz -lm -lm -lexpat -lexpat Modules/_hacl/libHacl_Hash_SHA2.a -lz -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2   >&5
/usr/bin/ld: cannot find Modules/_hacl/libHacl_Hash_SHA2.a: No such file or directory
collect2: error: ld returned 1 exit status
```

It seems this compiled fine on 22.04 without libHacl_Hash_SHA2.a also so maybe something changed in the configure file?

**Comment by Julien-Elie** [comment]
Have you really run `./autogen` as said in my response to your previous ticket #301?

The output you quoted is not the one that would be generated by the current Git source.

**Comment by Julien-Elie** [comment]
For the record, this issue was fixed last February in INN (https://github.com/InterNetNews/inn/commit/b3abd6bbc6049385ec6130039f802da5ba5c6503) following Debian bug report [#1061315](https://bugs.debian.org/1061315) related to Ubuntu builds.  This was the experimental Noble version at that time (released as Ubuntu 24.04).  This is exactly the version on which you faced the issue.

As the fix is done on a file used at configure time, `./autogen` and `./configure` need being run after `git pull` so as to actually have the fix.

**Comment by Julien-Elie** [comment]
See #295.
